services/worker_service.rb

```diff
@@ -27,15 +28,18 @@ def start_worker_for_provider_credential_and_config(project: nil, provider_crede
       raise "incompatible repo_config and provider_credential" if provider_credential.type != repo_config.provider_credential_type_needed
 
       logger.debug("Starting worker for #{project.project_name}, on behalf of #{user_responsible.email}")
-      new_worker = nil
+      new_workers = []
       case provider_credential.type
       when FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github]
-        new_worker = FastlaneCI::CheckForNewCommitsOnGithubWorker.new(provider_credential: provider_credential, project: project)
+        new_workers.push(FastlaneCI::CheckForNewCommitsOnGithubWorker.new(provider_credential: provider_credential, project: project))
+        new_workers.push(FastlaneCI::NightlyBuildGithubWorker.new(provider_credential: provider_credential, project: project))
       else
         raise "unrecognized provider_type: #{provider_credential.type}"
       end
 
-      self.resource_workers_dictionary[worker_key] = new_worker
+      new_workers.each do |new_worker|
+        self.resource_workers_dictionary[worker_key] = new_worker
```

Comment:
if there are two workers in `new_workers`, this will only store the last one since we're using the same `worker_key`. I think you need to generate a new key for each worker

Reply:
Alternatively, you could make it so that `resource_workers_dictionary` is a hash of `{ key => workers_array }` instead of `{key => single_worker}`